iroha/src/asset.rs

```diff
@@ -6,101 +6,103 @@ use parity_scale_codec::{Decode, Encode};
-    pub struct CreateAsset {
-        pub asset_name: String,
-        pub domain_id: String,
-        pub decimals: u8,
+impl From<&str> for Id {
+    fn from(string: &str) -> Id {
+        let vector: Vec<&str> = string.split('@').collect();
+        Id {
+            name: String::from(vector[0]),
+            container: String::from(vector[1]),
+            account: String::from(vector[2]),
+        }
     }
+}
+
+impl Identifiable for Asset {
+    type Id = Id;
+}
+
+pub mod isi {
+    use super::*;
+    use crate::isi::Mint;
+    use iroha_derive::*;
 
-    /// The purpose of transfer asset command is to share assets within the account in peer
-    /// network: in the way that source account transfers assets to the target account.
-    #[derive(Clone, Debug, PartialEq, Io, IntoContract, Encode, Decode)]
-    pub struct TransferAsset {
-        pub source_account_id: Id,
-        pub destination_account_id: Id,
-        pub asset_id: Id,
-        pub description: String,
-        pub amount: u128,
+    #[derive(Clone, Debug, Io, Encode, Decode)]
+    pub enum AssetInstruction {
+        MintAsset(u128, <Asset as Identifiable>::Id),
     }
 
-    impl Instruction for TransferAsset {
-        fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
-            let asset = world_state_view
-                .account(&self.source_account_id)
-                .ok_or("Source account not found")?
-                .assets
-                .get_mut(&self.asset_id)
-                .ok_or("Asset not found")?;
-            asset.amount -= self.amount;
-            let destination_account = world_state_view
-                .account(&self.destination_account_id)
-                .ok_or("Destionation account not found")?;
-            match destination_account.assets.get_mut(&self.asset_id.clone()) {
-                Some(asset) => {
-                    asset.amount += self.amount;
-                }
-                None => {
-                    destination_account.assets.insert(
-                        self.destination_account_id.clone(),
-                        Asset::new(self.asset_id.clone()).with_amount(self.amount),
-                    );
+    impl AssetInstruction {
+        pub fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
+            match self {
+                AssetInstruction::MintAsset(quantity, asset_id) => {
+                    Mint::new(*quantity, asset_id.clone()).execute(world_state_view)
                 }
             }
+        }
+    }
+
+    impl Mint<Asset, u128> {
+        pub fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
+            world_state_view
+                .asset(&self.destination_id)
+                .ok_or("Failed to find asset.")?
+                .quantity += self.object;
             Ok(())
         }
     }
+
+    impl From<Mint<Asset, u128>> for Instruction {
```

Comment:
The opposite of mint is to demint, so I think it is also needed.

Reply:
Sure, this is the minimum set of instructions in this PR